src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,288 @@
+                    this.props.store.selectedMolecularProfiles.result!,
+                    alterationData,
+                    true,
+                    this.props.store.genes.result!
+                );
+
+                if (data) {
+                    alterationFrequencyData.push(data);
+                }
+            }
+        );
+
+        return alterationFrequencyData;
+    }
+
+    @computed get alterationFrequencyDataForNonQueryGenes() {
+        const alterationFrequencyDataForNewGenes: ICBioData[] = [];
+
+        if (this.isNewStoreReady) {
+            this.storeForAllData!.oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.result!.forEach(
+                alterationData => {
+                    const data = getAlterationData(
+                        this.storeForAllData!.samples.result,
+                        this.storeForAllData!.patients.result,
+                        this.storeForAllData!.coverageInformation.result,
+                        this.storeForAllData!.sequencedSampleKeysByGene.result!,
+                        this.storeForAllData!.sequencedPatientKeysByGene
+                            .result!,
+                        this.storeForAllData!.selectedMolecularProfiles.result!,
+                        alterationData,
+                        false,
+                        this.props.store.genes.result!
+                    );
+
+                    if (data) {
+                        alterationFrequencyDataForNewGenes.push(data);
+                    }
+                }
+            );
+        }
+
+        // on pathway change PathwayMapper returns only the genes that are new (i.e genes for which we haven't
+        // calculated the alteration data yet), so we need to accumulate the alteration frequency data after each
+        // query
+        this.accumulatedAlterationFrequencyDataForNonQueryGenes = this.accumulatedAlterationFrequencyDataForNonQueryGenes.concat(
+            alterationFrequencyDataForNewGenes
+        );
+
+        return this.accumulatedAlterationFrequencyDataForNonQueryGenes;
+    }
+
+    @computed get isNewStoreReady() {
+        return (
+            this.storeForAllData &&
+            this.storeForAllData
+                .oqlFilteredCaseAggregatedDataByUnflattenedOQLLine.isComplete &&
+            this.storeForAllData.samples.isComplete &&
+            this.storeForAllData.patients.isComplete &&
+            this.storeForAllData.coverageInformation.isComplete &&
+            this.storeForAllData.sequencedSampleKeysByGene.isComplete &&
+            this.storeForAllData.sequencedPatientKeysByGene.isComplete &&
+            this.storeForAllData.selectedMolecularProfiles.isComplete
+        );
+    }
+
+    public render() {
+        // Alteration data of non-query genes are loaded.
+        if (this.isNewStoreReady) {
+            this.addGenomicData(this.alterationFrequencyData);
+            // Toasts are removed with delay
```

Comment:
can you denote somewhere that a toast is a notification? or is this an idiom I dont know 😁

Reply:
it's a little confusing to me the way that you're adding data and setting up the store in `render`

Reply:
I don't think `toast` is an idiom, it is the name of the component coming from `react-toastify` :)

The logic around the `addGenomicData` is confusing in general, I agree. Probably there is a cleaner way to do this. Also there is a not-yet-resolved discussion form the previous PR: https://github.com/cBioPortal/cbioportal-frontend/pull/2809#discussion_r352906940. We should probably refactor this `addGenomicData` mechanism.